doc/man/8/ceph.rst

```diff
@@ -928,6 +928,16 @@ Usage::
 Subcommand ``pool`` is used for managing data pools. It uses some additional
 subcommands.
 
+Subcommand ``application enable`` enables the pool for one of the applicatons -  object, block or file storage.
+
+Usage::
+        ceph osd pool application enable {pool-name} {rgw|rbd|cephfs}
```

Comment:
i think this usage is sort of misleading. because the app option can be a random string whose length is less or equal to 128. by looking at this document, i'd think that the app should be any of "rgw", "rbd" and "cephfs", but it cannot be one  specified by user, like "mails".

Reply:
what is application subcommand exactly used for? on i finding it here[1] i thought it performed some processing (on the pool or elsewhere) depending on type of storage. is it only like an annotation?

[1] http://docs.ceph.com/docs/master/rados/operations/pools/

Reply:
>  i thought it performed some processing (on the pool or elsewhere) depending on type of storage. 

AFAICT, currently only cephfs takes the tag into consideration when handling "fs authorize" command. it grants the access privilege to the pool tagged with "cephfs". but i think, in future, cephfs, rgw and rbd will also use their tags to perform more "processing" as you mentioned.

> is it only like an annotation?

yes. but the rados applications, cephfs, rgw and rbd, for instance, are, or will be, using it.

Reply:
Ok. But I do not understand that why would something like "mails" be useful since we have only 3 applications (cephfs, rbd, rgw) for a pool?

Or maybe, since it is possible to write something else after all, I _think_ it will be better to add a line (saying that it can hold other strings too) in explanation but not in the usage since other usages are not meaningful.

Reply:
@rishabh-d-dave 

> since we have only 3 applications (cephfs, rbd, rgw) for a pool?

if a pool is used for the backend storage for cephfs, then it has only 1 application. if the very pool is shared by cephfs, rbd and rgw, then there are 3 applications.

but if a user developed an application using librados for storing the mails and their attachments in certain pools, and in addition to the mail storage, he/she is also using the cluster for storing the vm images and for rgw in other pools. so in that case, to ease the ops' work, he/she might want to tag the pools used for mail storage.

Reply:
Got it.  Thanks for the explanation.  :)